api/src/main.js

```diff
@@ -4,66 +4,104 @@ const Koa = require('koa');
 const bodyParser = require('koa-bodyparser');
 const cors = require('@koa/cors');
 const responseTime = require('koa-response-time');
+const session = require('koa-session');
 const winston = require('winston');
 const config = require('./config');
 const { InputOutput } = require('./io');
+const passport = require('./middlewares/passport-config');
 
 winston.add(new winston.transports.Console());
 
 // Catch the uncaught errors that weren't wrapped in a domain or try catch statement
 process.on('uncaughtException', e => {
-    // This should never be called, as we handle all errors insides promises.
-    console.log('uncaughtException!!!', e);
-    // winston.log('error', e.message);
-    process.exit(1);
+  // This should never be called, as we handle all errors insides promises.
+  console.log('uncaughtException!!!', e);
+  // winston.log('error', e.message);
+  process.exit(1);
 });
 
 async function start() {
-    const app = new Koa();
-    app.context.io = new InputOutput();
-    await app.context.io.connect();
+  const app = new Koa();
+  app.context.io = new InputOutput();
+  await app.context.io.connect();
 
-    app.use(cors());
-    app.use(responseTime());
-    app.use(bodyParser({ enableTypes: ['json'] }));
-    app.use(require('./middlewares/error-handler'));
-    app.use(require('./routers/health').routes());
-    app.use(require('./middlewares/load-profile'));
-    app.use(require('./routers/downloads').routes());
-    app.use(require('./routers/invitations').routes());
-    app.use(require('./routers/input').routes());
-    app.use(require('./routers/project').routes());
-    app.use(require('./routers/rpc').routes());
-    app.use(require('./routers/uploads').routes());
+  // Store IO globally for passport strategies
+  global.io = app.context.io;
```

Comment:
no need for a global
you can wrap the passport config in a function, and provide `io` as a paremeter